### PR TITLE
iOS: automate version tagging and publishing

### PR DESCRIPTION
**Pull request by bangtoven**
### _How did you test your changes?_

manually

### CODE REVIEW

File: .github/workflows/ios-tagging.yml [hunk 1]
```diff
@@ -1,42 +1,18 @@
...
-        id: tag
-        run: |
-          COMMIT_MSG="${{ github.event.head_commit.message }}"
-          TAG=$(echo $COMMIT_MSG | sed -e '1!d' -e 's/\^ios-v([[:digit:]]*.[[:digit:]]*.[[:digit:]]*\).*$/\1/' )
-          echo "PARSED_TAG=$TAG" >> $GITHUB_ENV
-
-      - name: Tag
-        uses: actions/github-script@v6
+      - uses: actions/checkout@v2
+      - uses: ButlerLogic/action-autotag@1.1.1
```
Comment (bangtoven) [inline .github/workflows/ios-tagging.yml:13]: https://github.com/coinbase/coinbase-wallet-sdk/blob/master/.github/workflows/tagging.yml#L13

File: .github/workflows/ios-tagging.yml [hunk 2]
```diff
@@ -1,42 +1,59 @@
...
         uses: actions/checkout@v3
 
-      - name: Setup Tag Match
-        id: check-tag
-        run: |
-          if [[ "${{ github.event.head_commit.message }}" =~ ^ios-v[0-9]+\.[0-9]+\.[0-9]+ ]]; then
-            echo ::set-output name=match::true
-          fi
+      - id: autotag
+        uses: ButlerLogic/action-autotag@1.1.1
```
Comment (bangtoven) [inline .github/workflows/ios-tagging.yml:31]: we use the same action on the web sdk 
https://github.com/coinbase/coinbase-wallet-sdk/blob/master/.github/workflows/tagging.yml#L13

File: .github/workflows/ios-tagging.yml [hunk 3]
```diff
@@ -1,42 +1,59 @@
...
-        if: steps.check-tag.outputs.match == 'true'
-        id: tag
-        run: |
-          COMMIT_MSG="${{ github.event.head_commit.message }}"
-          TAG=$(echo $COMMIT_MSG | sed -e '1!d' -e 's/\^ios-v([[:digit:]]*.[[:digit:]]*.[[:digit:]]*\).*$/\1/' )
-          echo "PARSED_TAG=$TAG" >> $GITHUB_ENV
+  deploy:
+    name: Deploy
+    needs: autotag
+    if: ${{ needs.autotag.outputs.tagcreated == 'yes' }}
```
Comment (bangtoven) [inline .github/workflows/ios-tagging.yml:42]: only when new tag gets created
image

File: .github/workflows/ios-tagging.yml [hunk 4]
```diff
@@ -1,42 +1,59 @@
...
-              ref: "refs/tags/${{ env.PARSED_TAG }}",
-              sha: context.sha
-            })
+          ruby-version: '2.7.5'
+          bundler-cache: true
+      
+      - name: Deploy to Cocoapods
+        run: |
+          set -eo pipefail
+          pod trunk push CoinbaseWalletSDK.podspec --allow-warnings
```
Comment (bangtoven) [inline .github/workflows/ios-tagging.yml:57]: image

File: .github/workflows/ios-tagging.yml (hunk 3 above)
Comment (bangtoven) [reply .github/workflows/ios-tagging.yml:42]: image

**Review by AMITGOELNY — APPROVED** [review]

**Pull request merged**